### ISSUE TITLE
Can not attach Aria Label to button.

### ISSUE THREAD

**Issue by JakeLaCombe** [cerner/terra-core#837]
# Issue Description
For Terra Button, there doesn't look like it has an easy way to attach aria-label to the button. Given that terra-icon has the prop ariaLabel, buttons should have that same interface as well.

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [x] Bug
- [ ] Other

## Expected Behavior
ariaLabel is a prop on button

## Current Behavior
ariaLabel is not a prop on button

**Comment by mhemesath** [comment]
Shouldn't this be handled with custom props?

**Comment by neilpfeiffer** [comment]
Closing as already available in current API
